src/common/BaseAnnotator.ts

```diff
@@ -71,12 +74,26 @@ export default class BaseAnnotator {
         this.removeListener(LegacyEvent.SCALE, this.handleScale);
         this.removeListener(Event.ACTIVE_SET, this.setActiveAnnotationId);
         this.removeListener(Event.VISIBLE_SET, this.setVisibility);
+        document.removeEventListener('click', this.handleClick);
     }
 
     getElement(selector: HTMLElement | string): HTMLElement | null {
         return typeof selector === 'string' ? document.querySelector(selector) : selector;
     }
 
+    handleClick({ target }: MouseEvent): void {
+        const targetElement = target as HTMLElement;
+
+        if (
+            // If click target is not a part of the annotated element
+            (this.rootEl && !this.rootEl.contains(targetElement)) ||
+            // OR click target is part of the annotated element and not an annotation target or popup
+            !findClosestElWithClass(targetElement, ['ba-AnnotationTarget', 'ba-RegionAnnotations-popup'])
```

Comment:
I would very much like to avoid exposing the internals of the components inside the base annotator.